docs/design/2022-01-20-ticdc-mq-sink-multiple-topics.md

```diff
@@ -0,0 +1,146 @@
+# TiCDC supports multi-topic dispatch
+
+- Author(s): [hi-rustin](https://github.com/hi-rustin)
+- Tracking Issue: TODO: ADD IT BEFORE MERGE
+
+## Table of Contents
+
+- [Introduction](#introduction)
+- [Motivation or Background](#motivation-or-background)
+- [Detailed Design](#detailed-design)
+- [Test Design](#test-design)
+  - [Functional Tests](#functional-tests)
+  - [Scenario Tests](#scenario-tests)
+  - [Compatibility Tests](#compatibility-tests)
+  - [Benchmark Tests](#benchmark-tests)
+- [Impacts & Risks](#impacts--risks)
+- [Investigation & Alternatives](#investigation--alternatives)
+- [Unresolved Questions](#unresolved-questions)
+
+## Introduction
+
+This document provides a complete design on implementing multi-topic support in TiCDC MQ Sink.
+
+## Motivation or Background
+
+TiCDC MQ Sink only supports sending messages to a single topic, but in the MQ Sink usage scenario, we send data to
+systems like [Flink], which requires us to support multiple topics, each topic as a data source.
+
+## Detailed Design
+
+This solution will introduce a new configuration to the configuration file that specifies which topic the sink will send
+the table data to.
+
+We will continue to keep the original topic configuration in the sinkURI, which serves two purposes.
+
+1. when there is no new configuration or the configuration does not match, the data will be sent to that default topic.
+2. DDLs of the schema level will be sent to this topic by default.
+
+### Topic dispatch configuration format
+
+This configuration will be added to the TiCDC changefeed configuration file.
+
+```toml
+[sink]
+dispatchers = [
+  { matcher = ['test1.*', 'test2.*'], dispatcher = "ts", topic = "Topic dispatch expression" },
+  { matcher = ['test3.*', 'test4.*'], dispatcher = "rowid", topic = "Topic dispatch expression" },
+  { matcher = ['test1.*', 'test5.*'], dispatcher = "ts", topic = "Topic dispatch expression" },
+]
+```
+
+A new topic field has been added to dispatchers that will specify the topic dispatching rules for these tables.
+
+### Topic dispatch expression details
+
+The expression format looks like `flink_{schema}{table}`. This expression consists of two keywords and the `flink_`
+prefix.
+
+Two keywords:
+
+| Keyword  | Description            | Required |
+| -------- | ---------------------- | -------- |
+| {schema} | the name of the schema | no       |
+| {table}  | the name of the table  | no       |
+
+> When neither keyword is filled in, it is equivalent to sending the data to a fixed topic.
+
+`flink_` is the user-definable part, where the user can fill in the expression with a custom string.
+
+Some examples:
+
+```toml
+[sink]
+dispatchers = [
+  { matcher = ['test1.table1', 'test2.table1'], topic = "{schema}_{table}" },
+  { matcher = ['test3.*', 'test4.*'], topic = "flink_{schema}" },
```

Comment:
Is `underscore` a must separator, Is `topic-{schema}-{table}` valid?

Reply:
No. Let me modify this example.